newIDE/app/src/EventsBasedBehaviorEditor/EventsBasedBehaviorPropertiesEditor.js

```diff
@@ -154,6 +155,50 @@ export default function EventsBasedBehaviorPropertiesEditor({
 
   const forceUpdate = useForceUpdate();
 
+  const [searchText, setSearchText] = React.useState<string>('');
+  const [
+    searchMatchingPropertyNames,
+    setSearchMatchingPropertyNames,
+  ] = React.useState<Array<string>>([]);
+
+  const triggerSearch = React.useCallback(
+    () => {
+      const matchingProperties = mapVector(
+        properties,
+        (property: gdNamedPropertyDescriptor, i: number) => {
+          const lowerCaseSearchText = searchText.toLowerCase();
+          return property
+            .getName()
+            .toLowerCase()
+            .includes(lowerCaseSearchText) ||
+            property
+              .getLabel()
+              .toLowerCase()
+              .includes(lowerCaseSearchText) ||
+            property
+              .getGroup()
+              .toLowerCase()
+              .includes(lowerCaseSearchText)
+            ? property.getName()
+            : null;
+        }
+      ).filter(Boolean);
+      setSearchMatchingPropertyNames(matchingProperties);
+    },
+    [properties, searchText]
```

Comment:
I'm concerned that if you remove a property, this will render again but `properties` will still be the same C++ object, searchText will be the same, and you will end up with `searchMatchingPropertyNames` being an array that points to some `property` that are now pointing to delete objects.

Reply:
`searchMatchingPropertyNames` is used as a filter not an iterator, it should be fine for a deletion.
When a new property is added, the search is clear (it should probably be done when a property is pasted too).

Reply:
Still, I wonder if it's just safer to not store this in a state and just recompute it at every render?

> is used as a filter not an iterator

Sure, but there is still the risk that if something else does a change to `properties` outside, even a re-render of this component won't work because `properties` is unchanged (C++ object, it's always the same), searchText was forgotten to be modified (or was empty and is still empty - like, if I delete a property and there was no search text at all), and we end up with a property pointing to something deleted/invalid.

I say this because I already seen bugs for properties that were not valid anymore.

Reply:
Note that `matchingProperties` should be called `matchingPropertieNames`. There are only `string` in the state.

Reply:
Aaaah I totally missed that it's just strings 🤦‍♂️ All good then!